File: leetcode226.cs
public class TreeNode {
  public int val;
  public TreeNode? left;
  public TreeNode? right;

  public TreeNode(int val=0, TreeNode? left=null, TreeNode? right=null) {
    this.val = val;
    this.left = left;
    this.right = right;
  }
}

public class Program {

  // Remove static keyword if copy-pasted to Leetcode.
  public static TreeNode? InvertTree(TreeNode? node) {
    if (node == null) return node;
    (node.right, node.left) = (node.left, node.right);
    InvertTree(node.right);
    InvertTree(node.left);
    return node;
  }

  public static void Main() {
    TreeNode node1 = new TreeNode(1);
    TreeNode node3 = new TreeNode(3);
    TreeNode node2 = new TreeNode(2, node1, node3);
    TreeNode node6 = new TreeNode(6);
    TreeNode node9 = new TreeNode(9);
    TreeNode node7 = new TreeNode(7, node6, node9);
    TreeNode root = new TreeNode(4, node2, node7);
    InvertTree(root);
  }

}

File: leetcode226.py
# pylint: disable=C0115, C0116, C0103
"""
[*] DIFFICULTY: Easy
[*] QUESTION: 226. Invert Binary Tree

[*] DESCRIPTION:
Given the root of a binary tree, invert the tree, and return its root.

[*] INTUITION/APPROACH:
[*] TIME COMPLEXITY: O(n), n = amount of nodes
[*] THOUGHTS:
"""
from typing import Optional

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def __recursive(self, node: Optional[TreeNode]):
        node.left, node.right = node.right, node.left
        if node.right:
            self.__recursive(node.right)
        if node.left:
            self.__recursive(node.left)

    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root:
            self.__recursive(root)
        return root

    # used for testing
    def __call__(self, params):
        method = [m for m in dir(self) if callable(getattr(self, m)) and not m.startswith("_")]
        assert len(method) == 1, f"Only one public method allowed. Found {len(method)}"
        return getattr(self, method[0])(*params)


solution = Solution()
# used for testing
def test(params, answer):
    result = solution(params)
    if result != answer:
        print(f"Got {result} expected {answer} with params:", *params)


if __name__ == "__main__":
    pass
